### DISCUSSION TITLE
请教一些问题

### DISCUSSION THREAD

**Discussion by ListenV** [spencerwooo/onedrive-vercel-index#575] [Q&A] [answered]
1. 请问如果使用官方的API的话（就是官方提供的clientId与obfuscatedClientSecret），会影响到微软E5的续订吗？
2. 如果使用自己的API的话，AES加密的密钥可以不使用默认的 `onedrive-vercel-index` 吗？

**Comment by Android-KitKat** [comment]
2. AES 加密的密钥是在此处定义的。
https://github.com/spencerwooo/onedrive-vercel-index/blob/f7732101e8e495dbd3d6bfb64c3e3c3fb67a5314/utils/oAuthHandler.ts#L8

**Reply by ListenV** [reply]
了解了

**Comment by spencerwooo** [comment, marked as answer]
1. 本项目并不一定能保证 E5 开发者的续订，不论是用我们提供的 secret 还是自己的
2. 如上，可以自己改

**Reply by ListenV** [reply]
第二条，如果自己改的话，那仓库是不是要改为Private？

**Reply by ListenV** [reply]
> 第二条，如果自己改的话，那仓库是不是要改为Private？

好吧，我问的是废话

**Reply by spencerwooo** [reply]
没有，实际上是：无论你加密与否，secret 都不重要 —— 即使 obfuscated 的 secret 被攻击者反推回去拿到原 secret，攻击者也无法利用这一对 secret 做任何有价值的事情。攻击者只能「仿造身份」，即：以 secret 的拥有者的身份来创建自己的服务等。

**Reply by spencerwooo** [reply]
如果实在担心，可以将这一 secret 放在 Vercel 的环境变量里。

**Reply by ListenV** [reply]
明白了，感谢解惑😀
